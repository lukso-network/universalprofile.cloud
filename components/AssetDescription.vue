<script setup lang="ts">
type Props = {
  asset?: Asset
  withoutTitle?: boolean
}

const props = defineProps<Props>()
const { formatMessage } = useIntl()
const isLoaded = computed(() => props.asset && !props.asset?.isMetadataLoading)
const description = computed(() => props.asset?.resolvedMetadata?.description)
</script>

<template>
  <template v-if="isLoaded">
    <div v-if="description">
      <div v-if="!withoutTitle" class="heading-inter-14-bold pb-3">
        {{ formatMessage('token_details_description') }}
      </div>
      <div class="paragraph-inter-12-regular whitespace-pre-line break-word">
        {{ description }}
      </div>
    </div>
  </template>
  <AppPlaceholderMultiline v-else />
</template>
