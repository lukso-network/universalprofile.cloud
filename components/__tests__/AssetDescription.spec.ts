import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import AssetDescription from '@/components/AssetDescription.vue'

describe('AssetDescription', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render loading state', async () => {
    const component = await renderSuspended(AssetDescription)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render description', async () => {
    const component = await renderSuspended(AssetDescription, {
      props: {
        asset: {
          resolvedMetadata: {
            description: 'Lorem ipsum dolor sit amet',
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render for no description', async () => {
    const component = await renderSuspended(AssetDescription, {
      props: {
        asset: {
          resolvedMetadata: {
            description: undefined,
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render without title', async () => {
    const component = await renderSuspended(AssetDescription, {
      props: {
        withoutTitle: true,
        asset: {
          resolvedMetadata: {
            description: 'Lorem ipsum dolor sit amet',
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
