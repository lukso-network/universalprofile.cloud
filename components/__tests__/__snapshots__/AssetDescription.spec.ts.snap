// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetDescription > should render description 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="heading-inter-14-bold pb-3">Token Description</div>
    <div class="paragraph-inter-12-regular whitespace-pre-line break-word">Lorem ipsum dolor sit amet</div>
  </div>
</div>"
`;

exports[`AssetDescription > should render for no description 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`AssetDescription > should render loading state 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="mb-3 h-[17px] w-1/2 animate-pulse rounded-4 bg-neutral-90"></div>
    <div class="">
      <div class="grid gap-1">
        <div class="animate-pulse rounded-4 bg-neutral-90 h-[17px] w-1/2"></div>
        <div class="animate-pulse rounded-4 bg-neutral-90 h-[17px] w-4/5"></div>
        <div class="animate-pulse rounded-4 bg-neutral-90 h-[17px] w-3/5"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AssetDescription > should render without title 1`] = `
"<div id="test-wrapper">
  <div>
    <!--v-if-->
    <div class="paragraph-inter-12-regular whitespace-pre-line break-word">Lorem ipsum dolor sit amet</div>
  </div>
</div>"
`;
